stake-pool/program/src/processor.rs

```diff
@@ -2005,9 +2007,7 @@ impl Processor {
             program_id,
             stake_pool_info.key,
         )?;
-        if let Some(sol_deposit_authority) = sol_deposit_authority_info {
-            stake_pool.check_sol_deposit_authority(sol_deposit_authority)?;
-        }
```

Comment:
This was only doing the check if the last account was passed, which means that it was possible to bypass it totally if you just didn't pass the last account.

Reply:
Oh man, that's a great catch, thanks